dracut.sh

```diff
@@ -1107,23 +1115,26 @@ if ! [[ $outfile ]]; then
             | while read -r line || [[ $line ]]; do
                 [[ $line =~ BUILD_ID\=* ]] && eval "$line" && echo "$BUILD_ID" && break
             done)
-        if [[ -z $dracutsysrootdir ]]; then
-            if [[ -d /efi ]] && mountpoint -q /efi; then
-                efidir=/efi/EFI
+        if [[ -z $uefi_outdir ]]; then
+            if [[ -z $dracutsysrootdir ]]; then
+                if [[ -d /efi ]] && mountpoint -q /efi; then
+                    uefi_outdir=/efi/EFI
+                else
+                    uefi_outdir=/boot/EFI
+                    if [[ -d /boot/efi/EFI ]]; then
+                        uefi_outdir=/boot/efi/EFI
+                    fi
+                fi
             else
-                efidir=/boot/EFI
-                if [[ -d /boot/efi/EFI ]]; then
-                    efidir=/boot/efi/EFI
+                uefi_outdir=/boot/EFI
+                if [[ -d $dracutsysrootdir/boot/efi/EFI ]]; then
+                    uefi_outdir=/boot/efi/EFI
                 fi
             fi
-        else
-            efidir=/boot/EFI
-            if [[ -d $dracutsysrootdir/boot/efi/EFI ]]; then
-                efidir=/boot/efi/EFI
-            fi
+            uefi_outdir="$uefi_outdir/Linux"
```

Comment:
I would like to understand this line better. Why is "Linux" appended to uefi_outdir here ?

Reply:
It's what currently dracut is doing by default since https://github.com/dracutdevs/dracut/commit/5c57209b. See https://systemd.io/BOOT_LOADER_SPECIFICATION/#type-2-efi-unified-kernel-images

Reply:
Hmm, I think the GitHub diff is confusing you, the "/Linux" part is only appended if the user does not set a custom uefi output directory. See the snippet as a whole:

https://github.com/dracutdevs/dracut/blob/65ae8d8d0d2148a5a6e6a31adb355093c8915675/dracut.sh#L1118-L1137